<launch>
	<node pkg="rviz" type="rviz" name="visualization"/>
	
	<group ns="fmDecisionMakers">
		<node pkg="sdu_frobit_casmo" name="mission" type="casmo.py" output="screen">
    		<param name="deadman_topic" value="/fmSignals/deadman" />
			<param name="cmd_vel_topic" value="/fmSignals/cmd_vel" />
			<param name="odom_topic" value="/fmKnowledge/odom" />
			<param name="odom_frame" value="/odom_combined"/> 
			<param name="base_frame" value="/base_link" />
			<param name="max_linear_velocity" value="0.4"/>
			<param name="max_angular_velocity" value="1.8"/>
			<param name="deadband" value="10"/>
			<param name="reduced_range" value="40"/>
			<param name="feedback_topic" value="/joy/set_feedback"/>
			<param name="joy_topic" value="/joy" />
			<param name="status_topic" value="/fmData/status"/>
			<param name="publish_frequency" value="10"/>
		</node>
	</group>
	
        <node pkg="wiimote" type="wiimote_node.py" name="remote_control" respawn="true">
        	<param name="fixed_wiimote_address" value="00:1F:32:B3:A2:F3"/>
		</node>
		
	<group ns="platform_executors">
		<node pkg="line_action_server" name="lineActionServer" type="lineGoalActionServer.py" output="screen">
			<param name="cmd_vel_topic" value="/fmSignals/cmd_vel" />
			<param name="odometry_topic" value="/fmKnowledge/odom" /> 
			<param name="odom_frame" value="/odom_combined"/> 
			<param name="use_tf" value="True"/> 
			<param name="max_linear_velocity" value="0.4"/> 
			<param name="max_angular_velocity" value="0.8"/>
			<param name="max_distance_error" value="0.20"/> 
			<param name="max_distance_from_line" value="0.2"/>
			<param name="retarder" value="0.8"/> 
			<param name="max_angle_error" value="0.5"/> 
			<param name="max_initial_error" value="0.1"/> 
			<param name="period" value="0.1"/> 
			<param name="lin_p" value="0.4"/>
			<param name="lin_i" value="0.2"/> 
			<param name="lin_d" value="0.0"/> 
			<param name="ang_p" value="0.4"/> 
			<param name="ang_i" value="0.0"/> 
			<param name="ang_d" value="0.0"/> 
			<param name="integrator_max" value="0.1"/> 
		</node>
	</group>	

</launch>
